spec/models/assign_msa_component_to_service_spec.rb

```diff
@@ -37,6 +37,14 @@
     expect(event.errors.full_messages.first).to eq('Service Wrong component type')
   end
 
+  it 'deassociates msa component from the service' do
+    component = create(:msa_component)
+    event = create(:assign_msa_component_to_service_event, msa_component_id: component.id)
+    expect(event.service.msa_component_id).to eq(component.id)
+    DeleteComponentEvent.create(component: component)
+    expect(Service.find_by_id(event.service).msa_component_id).to be_nil
```

Comment:
to be 100% accurate, it probably should be this:
```suggestion
    expect(Service.find_by_id(event.service.id).msa_component_id).to be_nil
```
but it works :D